hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -181,6 +181,31 @@ public void incrementCount(List<Long> txIDs)
     }
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @throws IOException
+   */
+  @Override
+  public int resetCount(List<Long> txIDs) throws IOException {
+    List<Long> failedTransactions = getFailedTransactions().stream()
+        .map(DeletedBlocksTransaction::getTxID).collect(Collectors.toList());
+    if (txIDs != null && !txIDs.isEmpty()) {
+      failedTransactions = failedTransactions.stream().filter(txIDs::contains)
+          .collect(Collectors.toList());
+    }
+    lock.lock();
+    try {
+      return deletedBlockLogStateManager.resetRetryCountOfTransactionInDB(
+            new ArrayList<>(failedTransactions));
+    } catch (TimeoutException | IOException ex) {
+      LOG.error("Cannot reset block deletion transactions {}",
+          failedTransactions, ex);
+      return 0;
```

Comment:
I think the method should either declare `throws IOException` or catch it, but not both.

Reply:
I referred to the line in BlockManagerImpl:259,  both exceptions could be treated as the same, and the maintainer shall find the stack through the log to recognize the root cause. If it is still not a good idea, pls comment this again.

Reply:
Grouping `IOException` and `TimeoutException` in the same `catch` is not a problem.  Sorry if my original comment was not clear.

This method is declared as:

```java
int resetCount(List<Long> txIDs) throws IOException;
```

but it never throws `IOException`, since the method catches it and returns 0 instead.  So we can simplify the definition as:

```java
int resetCount(List<Long> txIDs);
```

Or we can keep `throws`, but drop `IOException` from the `catch`.

Reply:
Most excepetions in `DeletedBlockLogImpl` are not thrown, so I inherited this.